test/datadog/lambda/metrics.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockLength
+require 'datadog/lambda'
+require 'datadog/statsd'
+require 'datadog/lambda/metrics'
+require 'datadog/lambda/utils/logger'
+
+describe Datadog::Metrics::Client do
+  subject(:client) { Datadog::Metrics::Client.instance }
+  let(:layer_tag) { Datadog::Lambda.dd_lambda_layer_tag }
+  let(:default_tags) { ["dd_lambda_layer:datadog-ruby#{layer_tag}"] }
+
+  describe '.instance' do
+    it 'returns the same instance' do
+      instance1 = Datadog::Metrics::Client.instance
+      instance2 = Datadog::Metrics::Client.instance
+      expect(instance1).to be(instance2)
+    end
+  end
+
+  describe '#distribution' do
+    context 'when extension is running' do
+      let(:statsd) { instance_double(Datadog::Statsd) }
+
+      before(:each) do
+        # Stub the extension_running? method to return true
+        allow(Datadog::Utils).to receive(:extension_running?).and_return(true)
+
+        # Mock Datadog::Statsd client
+        @previous_statsd = client.instance_variable_get(:@statsd)
+        client.instance_variable_set(:@statsd, statsd)
+      end
+
+      after(:each) do
+        # Reset Datadog::Statsd mock
+        client.instance_variable_set(:@statsd, @previous_statsd)
+      end
+
+      it 'sends metrics properly' do
+        expected_tags = default_tags.concat(['env:dev', 'region:nyc'])
+        # Expect the distribution method to be called with the correct arguments
+        expect(statsd).to receive(:distribution).with('metric_name', 42, tags: expected_tags)
+
+        # Call the distribution method
+        client.distribution('metric_name', 42, env: 'dev', region: 'nyc')
+      end
+
+      it 'doesnt send metric when statsd fails' do
+        expect(statsd).to receive(:distribution).and_raise 'not sending metric!'
+
+        expected = 'error sending metric to the extension: not sending metric!'
+        expect(Datadog::Utils.logger).to receive(:warning).with(expected)
+
+        # Call the distribution method, which should raise the specified error
+        # and the logger should print it as debug
+        client.distribution('metric_name', 42, env: 'dev', region: 'nyc')
+      end
+    end
+
+    it 'prints metrics when extension is not running' do
```

Comment:
If think you should test two cases:
* When time is not provided, which you are testing here.
* When time is provided: we'd assert that we passes the right time through the statsd api.

Reply:
We currently ignore when time is passed if the extension (statsd client instantiated) is running. Since this is not implemented in the statsd clients.

Reply:
Tested in `lambda.rb`